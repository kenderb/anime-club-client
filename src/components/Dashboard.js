import React from 'react';
import { connect } from 'react-redux';
import PropType from 'prop-types';
import { Redirect } from 'react-router-dom';

const Dashboard = ({ user }) => {
  if (!user.loggedIn) return <Redirect to="/" />;
  return (
    <div>
      User logged In ?:
      {' '}
      {`${user.loggedIn}`}
      <h1>Hello from Dashboard component</h1>
    </div>
  );
};
const mapStateToprops = state => ({
  user: state.user,
});

Dashboard.propTypes = {
  user: PropType.instanceOf(Object).isRequired,
};
export default connect(mapStateToprops)(Dashboard);
