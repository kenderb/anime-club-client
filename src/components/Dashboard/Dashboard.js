import React, { useEffect, useState } from 'react';
import { connect } from 'react-redux';
import PropType from 'prop-types';
import { Redirect } from 'react-router-dom';
import AnimeCard from '../AnimeCard';
import { getFavoritesList, getAllAnimes } from '../../actions';
import './Dashboard.style.css';

const Dashboard = ({
  user, getFavoritesList, animes, getAllAnimes,
}) => {
  const [currentNumber, setCurrentUser] = useState(1);

  useEffect(() => {
    getFavoritesList({ userId: user.id });
    getAllAnimes();
  }, []);

  const handleCurrentSelect = id => {
    setCurrentUser(id);
  };
  if (!user.loggedIn) return <Redirect to="/login" />;
  return (
    <div>
      <h1 className="dashboard-title">
        Animes
      </h1>
      <ul className="anime-wrapper d-flex">
        {animes.map(anime => (
          <AnimeCard
            key={anime.id}
            anime={anime}
            onClickImage={() => handleCurrentSelect(anime.id)}
          />
        ))}
      </ul>
      <div className="anime-count-container">
        {currentNumber}
        /
        {animes.length}
      </div>
    </div>
  );
};
const mapStateToprops = state => ({
  user: state.user,
  animes: state.animes,
});

Dashboard.propTypes = {
  user: PropType.instanceOf(Object).isRequired,
  animes: PropType.instanceOf(Array),
  getFavoritesList: PropType.func.isRequired,
  getAllAnimes: PropType.func.isRequired,
};
Dashboard.defaultProps = {
  animes: [],
};
export default connect(mapStateToprops, { getFavoritesList, getAllAnimes })(Dashboard);
