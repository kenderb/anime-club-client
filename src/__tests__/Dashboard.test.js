import thunk from 'redux-thunk';
import { Provider } from 'react-redux';
import { create } from 'react-test-renderer';
import { createStore, applyMiddleware } from 'redux';
import { BrowserRouter } from 'react-router-dom';
import combineReducers from '../reducers';
import Dashboard from '../components/Dashboard';

test('Dashboard component renders correctly', () => {
  const store = createStore(combineReducers, applyMiddleware(thunk));
  const tree = create(
    <Provider store={store}>
      <BrowserRouter>
        <Dashboard />
      </BrowserRouter>
    </Provider>,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
