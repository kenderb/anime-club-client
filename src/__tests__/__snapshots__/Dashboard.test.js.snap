// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard component renders correctly 1`] = `null`;
